pennylane/transforms/control.py

```diff
@@ -105,8 +106,11 @@ def control_wires(self):
 
     def expand(self):
         tape = self.subtape
+        tape.set_parameters(self.data)

```

Comment:
How come this step was not being done already automatically before? For some reason, I'd think that since we have `self.subtape` ready, the trainable parameters would have been set already too. :thinking:

Reply:
Because the gradient rules use `Operation.data`, so it was setting `ControlledOperation.data`, but this was not affecting the subtape!

Reply:
Oh I see! :open_mouth: Great catch. :slightly_smiling_face: